pokedex-ruzala/src/components/EditPokemonModal.js

```diff
@@ -0,0 +1,253 @@
+import {
+  Box,
+  Button,
+  Modal,
+  Table,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableCell,
+  Typography,
+  TextField,
+  Select,
+  MenuItem,
+} from "@mui/material";
+import { useState, useContext } from "react";
+import { useSnackbar } from "notistack";
+import { useFormik } from "formik";
+import { pokemonEditSchema } from "../schemas";
+import { GlobalContext } from "../App";
+import modifyPokemon from "../functional/modifyPokemon";
+
+export default function EditPokemonModal({ pokemon }) {
+  const [open, setOpen] = useState(false);
+
+  const { enqueueSnackbar } = useSnackbar();
+  const { arrayOfModifiedPokemon, setArrayOfModifiedPokemon, pokemonTypes } =
+    useContext(GlobalContext);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const onSubmit = (values) => {
+    const newPokemon = { ...pokemon };
+    newPokemon.name = values.name.toLowerCase();
+    newPokemon.types[0].type.name = values.type1;
+    if (newPokemon.types[1]) {
+      newPokemon.types[1].type.name = values.type2;
+    }
+    newPokemon.weight = values.weight;
+    newPokemon.height = values.height;
+    newPokemon.base_experience = Number(values.base_experience);
+    console.log(typeof newPokemon.baseExperience);
+    modifyPokemon(
+      arrayOfModifiedPokemon,
+      setArrayOfModifiedPokemon,
+      newPokemon
+    );
+    enqueueSnackbar("Pomyślnie zmodyfikowano pokemona", { variant: "success" });
+  };
+
+  const { values, handleBlur, handleChange, handleSubmit, errors, touched } =
+    useFormik({
+      initialValues: {
+        name: `${pokemon.name}`,
+        type1: `${pokemon.types[0].type.name}`,
+        type2: pokemon.types[1] ? `${pokemon.types[1].type.name}` : undefined,
+        weight: `${pokemon.weight}`,
+        height: `${pokemon.height}`,
+        base_experience: `${pokemon.base_experience}`,
+      },
+      validationSchema: pokemonEditSchema,
+      onSubmit,
+    });
+
+  const style = {
+    position: "absolute",
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)",
+    width: "60%",
+    backgroundColor: "primary.light",
+    border: "2px solid black",
+    boxShadow: 24,
+  };
+  const scaling = 0.5;
+  return (
+    <Box>
+      <Button variant="contained" onClick={handleOpen}>
+        Edytuj
+      </Button>
+      <Modal open={open} onClose={handleClose}>
+        <Box sx={style}>
+          <Box
+            sx={{
+              display: "flex",
+              flexDirection: "column",
+              flexWrap: "wrap",
+              width: "100%",
+              justifyContent: "center",
+            }}
+          >
+            <Table sx={{ width: "80%" }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell>ID</TableCell>
```

Comment:
To powtarzających się elementów warto zadeklarować sobie tablicę a następnie po niej iterować